### PR TITLE
EES-1356 Add, Rename and Reorder methodology content accordion sections

### PR DESCRIPTION
**Pull request by bennettstuart**
CRU for content sections in methodology content

Delete isnt implemnted yet

### CODE REVIEW

File: src/explore-education-statistics-admin/src/pages/methodology/edit-methodology/content/MethodologyContentPage.tsx [hunk 1]
```diff
@@ -38,33 +46,122 @@ const MethodologyContentPage = ({
...
-          <h1
-            className="govuk-heading-xl"
-            data-testid={`page-title ${methodology.title}`}
-          >
-            {methodology.title}
-          </h1>
-          <p>Lorem ipsum.</p>
-          <p>{JSON.stringify(methodology)}</p>
-        </main>
+          <section className={isEditing ? 'page-editing' : 'page-preview'}>
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/methodology/edit-methodology/content/MethodologyContentPage.tsx]: Shouldn't these classes be namespaced i.e. `dfe-page-editing` and `dfe-page-preview`?

File: src/explore-education-statistics-admin/src/pages/methodology/edit-methodology/content/MethodologyContentPage.tsx [hunk 2]
```diff
@@ -38,33 +46,122 @@ const MethodologyContentPage = ({
...
+                      methodology.id,
+                      section.id as string,
+                      heading,
+                    )
+                  }
+                />
+              ))}
+            </Accordion>
+
+            <p>{JSON.stringify(methodology)}</p>
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/methodology/edit-methodology/content/MethodologyContentPage.tsx]: Is this WIP?

File: src/explore-education-statistics-admin/src/pages/methodology/edit-methodology/content/MethodologyContentPage.tsx [hunk 3]
```diff
@@ -38,33 +46,122 @@ const MethodologyContentPage = ({
...
+              onSaveOrder={order => {
+                return new Promise(resolve => {
+                  methodologyService
+                    .updateContentSectionsOrder(methodology.id, order)
+                    .then(resolve)
+                    .catch(handleApiErrors);
+                });
+              }}
+              onAddSection={() => {
+                return new Promise(resolve => {
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/methodology/edit-methodology/content/MethodologyContentPage.tsx]: Use async/await. This is currently dangerous, as if there is actually an error, the wrapping promise will never resolve.

File: src/explore-education-statistics-admin/src/pages/methodology/edit-methodology/content/MethodologyContentPage.tsx [hunk 4]
```diff
@@ -38,33 +46,122 @@ const MethodologyContentPage = ({
...
+                )}
+              </div>
+            </div>
+
+            <Accordion
+              id="contents-accordion"
+              canReorder
+              sectionName="Contents"
+              onSaveOrder={order => {
+                return new Promise(resolve => {
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/methodology/edit-methodology/content/MethodologyContentPage.tsx]: Use async/await. This is currently dangerous, as if there is actually an error, the wrapping promise will never resolve.

File: src/explore-education-statistics-admin/src/pages/methodology/edit-methodology/content/MethodologyContentPage.tsx (hunk 2 above)
Comment (bennettstuart) [reply src/explore-education-statistics-admin/src/pages/methodology/edit-methodology/content/MethodologyContentPage.tsx]: it was

**Pull request merged**